autoscale/src/main/java/com/sequenceiq/periscope/repository/ScalingTriggerRepository.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.periscope.repository;
+
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import javax.transaction.Transactional;
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+
+import com.sequenceiq.cloudbreak.workspace.repository.EntityType;
+import com.sequenceiq.periscope.api.model.TriggerStatus;
+import com.sequenceiq.periscope.domain.ScalingTrigger;
+
+@EntityType(entityClass = ScalingTrigger.class)
+@Transactional(Transactional.TxType.REQUIRED)
+public interface ScalingTriggerRepository extends CrudRepository<ScalingTrigger, Long> {
+
+    @Query("SELECT st FROM ScalingTrigger st WHERE st.triggerCrn = :triggerCrn")
+    Optional<ScalingTrigger> findByTriggerCrn(@Param("triggerCrn") String triggerCrn);
+
+    @Query("SELECT st FROM ScalingTrigger st WHERE st.cluster.id = :clusterId")
+    List<ScalingTrigger> findAllByCluster(@Param("clusterId") Long clusterId);
+
+    @Query("SELECT st FROM ScalingTrigger st WHERE st.cluster.id = :clusterId AND st.startTime <= :startTimeBefore")
+    List<ScalingTrigger> findAllByClusterWithStartTimeBefore(@Param("clusterId") Long clusterId, @Param("startTimeBefore") Date startTimeBefore);
+
+    @Query("SELECT st FROM ScalingTrigger st WHERE st.cluster.id = :clusterId AND st.startTime >= :startTimeAfter")
+    List<ScalingTrigger> findAllByClusterWithStartTimeAfter(@Param("clusterId") Long clusterId, @Param("startTimeAfter") Date startTimeAfter);
+
+    @Query("SELECT st FROM ScalingTrigger st WHERE st.cluster.id = :clusterId AND st.triggerStatus = :triggerStatus AND st.startTime >= :startTimeFrom " +
```

Comment:
Need to be a little careful with the boundary conditions here. This has the lower and upper bound for startTime marked as inclusive. Typically - one is inclusive and the other is exclusive, so that when run as a pagination request, or when run as a from X until Y followed by from Y until Z - we don't see duplicate results for Y.

Reply:
Updated the upper bound for these JPQL queries to be exclusive. More queries can be added based on the API endpoints being created as part of [CB-19544](https://jira.cloudera.com/browse/CB-19544)